homeassistant/components/mqtt/mixins.py

```diff
@@ -237,6 +239,33 @@ async def __call__(
         """Define setup_entities type."""
 
 
+async def async_get_platform_config_from_yaml(
+    hass: HomeAssistant, domain: str, schema: vol.Schema
+) -> list[ConfigType]:
+    """Return a list of validated configurations for the platform read from configuration.yaml."""
+
+    def check_schema(config: dict, schema: vol.Schema) -> ConfigType:
+        """Update the platform for schema compatibility and check the schema."""
+        if CONF_PLATFORM in config:
+            error_string = f"Invalid keyword 'platform' found, please remove it from your configuration: {json.dumps(config)}"
+            raise ValueError(error_string)
+        config[CONF_PLATFORM] = DOMAIN
```

Comment:
Instead of this:
- modify + rename schemas
`MQTT_BASE_PLATFORM_SCHEMA` + `MQTT_RO_PLATFORM_SCHEMA` + `MQTT_RW_PLATFORM_SCHEMA`:
  - Modify MQTT_BASE_PLATFORM_SCHEMA to not derive from `cv.PLATFORM_SCHEMA`
  - Rename to `MQTT_BASE_SCHEMA` + `MQTT_RO_SCHEMA` + `MQTT_RW_SCHEMA`
  - `_PLATFORM_SCHEMA_BASE` will now not include the `platform` key.
- Update `PLATFORM_SCHEMA` so it extends  `cv.PLATFORM_SCHEMA`
- Add a new `PLATFORM_SCHEMA_MODERN` or similar, which should be same as `DISCOVERY_SCHEMA` but without the `extra=vol.REMOVE_EXTRA` and pass that to `async_get_platform_config_from_yaml`

Reply:
The old MQTT_BASE_PLATFORM_SCHEMA + MQTT_RO_PLATFORM_SCHEMA + MQTT_RW_PLATFORM_SCHEMA schema's are still there for the platforms that I did not update yet.